pig/src/main/java/org/jboss/pnc/bacon/pig/impl/utils/FileUtils.java

```diff
@@ -570,4 +571,32 @@ public static void getFileFromZip(String zipName, String fileName, String extrac
             throw new RuntimeException("Unzip of " + fileName + " from " + zipName + " failed", e);
         }
     }
+
+    public static void replaceFileString(String orig, String replace, String fileName) {
+        try {
+            FileInputStream fis = new FileInputStream(fileName);
+            String content = org.apache.commons.io.IOUtils.toString(fis, Charset.defaultCharset());
+            content = content.replaceAll(orig, replace);
+            FileOutputStream fos = new FileOutputStream(fileName);
+            org.apache.commons.io.IOUtils.write(content, new FileOutputStream(fileName), Charset.defaultCharset());
+            fis.close();
+            fos.close();
```

Comment:
Convert all `try` statements to `try-with-resources` or `try-finally`, or else there is currently a resource leak any time an exception is thrown.

Reply:
In fact you could use `FileUtils.readFileToString` , `FileUtils.writeStringToFile` which wraps a lot of the above and would make it simpler.

Reply:
Sure, I will use these methods. I just relocated the original code to Util class since it was going to be used in more than one place

Reply:
Not just with files, though, but any `try` where you have to close resources.